src/AutoRest.CSharp/Common/Input/InputTypes/Serialization/TypeSpecInputOperationConverter.cs

```diff
@@ -85,17 +87,38 @@ public static InputOperation CreateOperation(ref Utf8JsonReader reader, string?
                 Console.Error.WriteLine($"[Warn]: InputOperation '{name}' does not have a description");
                 description = $"{name.Humanize()}.";
             }
+            crossLanguageDefinitionId = crossLanguageDefinitionId ?? throw new JsonException("InputOperation must have crossLanguageDefinitionId");
             uri = uri ?? throw new JsonException("InputOperation must have uri");
             path = path ?? throw new JsonException("InputOperation must have path");
             parameters = parameters ?? throw new JsonException("InputOperation must have parameters");
             responses = responses ?? throw new JsonException("InputOperation must have responses");
 
-            var inputOperation = new InputOperation(name, resourceName, summary, deprecated, description, accessibility, parameters, responses, httpMethod, requestBodyMediaType, uri, path, externalDocsUri, requestMediaTypes, bufferResponse, longRunning, paging, generateProtocolMethod, generateConvenienceMethod, keepClientDefaultValue);
+            var inputOperation = new InputOperation(name, resourceName, summary, deprecated, description, accessibility, parameters, responses, httpMethod, requestBodyMediaType, uri, path, externalDocsUri, requestMediaTypes, bufferResponse, longRunning, paging, generateProtocolMethod, generateConvenienceMethod, crossLanguageDefinitionId, keepClientDefaultValue)
+            {
+                IsNameChanged = IsNameChanged(crossLanguageDefinitionId, name)
```

Comment:
For Swagger input, `IsNameChanged` is always `false`. But we know we can change the name in other way. Are we diverse here?

Reply:
Yes and I think it is impossible to unify those because typespec will never have those configurations that allow us to override the operation name.

Reply:
I think we could merge these two ways of renaming operations when we fully moved to MGC by changing how it works on the swagger side